packages/core/src/elements-context.ts

```diff
@@ -0,0 +1,85 @@
+import { InternalSchemaElement } from './typeschema/types';
+import { getPathDifference } from './utils';
+
+export type ElementsContextType = {
+  path: string;
+  profileUrl: string | undefined;
+  elements: Record<string, InternalSchemaElement>;
```

Comment:
I need to dig into this one more... It seems like this would be duplicative to all of the existing code we have with `InternalTypeSchema.elements`

Or, in other words... is this something that could have been added to `InternalTypeSchema` directly to avoid creating more internal schema types?

Reply:
I added more documentation to ElementsContext like you suggested elsewhere would help clear this up. ElementsContext exists to normalize working with elements across resources, extensions, and slices...really just normalizing traversing values either outside of or within slices. I don't believe there's a straightforward change that could be made to `InternalTypeSchema` to achieve anything similar. All the functions like `getElementDefinition`, `getDataType` revolve around `InternalTypeSchema`, but that just fundamentally doesn't work in slices where there isn't a corresponding `InternalTypeSchema`.

We could do something like make a pseudo `InternalTypeSchema` for slices that also get injected into DATA_TYPES, `getDataType`, etc., but that would definitely be violating the abstraction and pretty blatantly lying to the computer.

Reply:
I think the eventual final form is that something like `ElementsContextType` is eventually merged into `InternalTypeSchema` and `SliceDefinition` to unify the two.